jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -186,6 +207,85 @@ async def job_script_update(job_submission_id: int, job_submission: JobSubmissio
     return job_submission_data
 
 
+# The "agent" routes are used for agents to fetch pending job submissions and update their statuses
+@router.get(
+    "/job-submissions/agent/pending",
+    description="Endpoint to list pending job_submissions for the requesting client",
+    response_model=List[PendingJobSubmission],
+)
+async def job_submissions_agent_pending(
+    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SUBMISSIONS_VIEW)),
+):
+    """
+    Get a list of pending job submissions for the cluster-agent.
+    """
+    if token_payload.client_id is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Access token does not contain a `client_id`. Cannot fetch pending submissions",
+        )
+
+    query = (
+        select(
+            columns=[
+                job_submissions_table.c.id,
+                job_submissions_table.c.job_submission_name,
+                job_scripts_table.c.job_script_name,
+                job_scripts_table.c.job_script_data_as_string,
+                applications_table.c.application_name,
+            ]
+        )
+        .select_from(job_submissions_table.join(job_scripts_table).join(applications_table))
+        .where(job_submissions_table.c.status == JobSubmissionStatus.CREATED,)
+        .where(job_submissions_table.c.cluster_client_id == token_payload.client_id,)
+    )
+
+    rows = await database.fetch_all(query)
+    return rows
+
+
+@router.put(
+    "/job-submissions/agent/{job_submission_id}",
+    status_code=200,
+    description="Endpoint for an agent to update the status of a job_submission",
+    response_model=JobSubmissionResponse,
+)
```

Comment:
I think this route should be a PATCH method and raise an error in case the job submissions ID doesn't match any real job submissions

Reply:
If the job_submission_id doesn't match an existing row, the update database query will nave a null result which causes a 404 being raised.

It does make sense for it to be a PATCH instead of PUT, however.

Reply:
I see. So what about just changing this one and change the others in the future?

Reply:
I will create a task to do this in a future sprint.